src/item.cpp

```diff
@@ -3983,10 +3966,10 @@ long item::ammo_required() const
     return 0;
 }
 
-bool item::ammo_consume( int qty, const tripoint& pos ) {
+long item::ammo_consume( long qty, const tripoint *pos ) {
```

Comment:
> except in process_tool where it's most likely a bug

Well spotted.

> I don't see you using it in any place where a sensible pos can't be supplied,

As it gradually becomes a replacement for manipulating `charges` directly the situation often recurs. I managed to work around it for #16049 by passing it from `player::use_charges`.

> will enforce passing around the position

Nothing to stop someone supplying `tripoint_min` instead of refactoring to be able to supply the correct position.

You are entirely correct however that lack of `pos` should be exceptional so the issue here is that the parameter shouldn't be defaulted.